boards/arduino/portenta_c33/eth_pwm_clock.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2025 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(eth_pwm_clock, CONFIG_CLOCK_CONTROL_LOG_LEVEL);
+
+int eth_pwm_clock_enable(void)
+{
+	int ret;
+	uint32_t rate;
+	const struct device *eth_pwm_clk_dev = DEVICE_DT_GET(DT_NODELABEL(pwmclock));
+
+	if (!device_is_ready(eth_pwm_clk_dev)) {
+		LOG_ERR("Camera external clock source device is not ready!");
+		return -ENODEV;
+	}
+
+	ret = clock_control_on(eth_pwm_clk_dev, (clock_control_subsys_t)0);
+	if (ret < 0) {
+		LOG_ERR("Failed to enable camera external clock error: (%d)", ret);
+		return ret;
+	}
+
+	ret = clock_control_get_rate(eth_pwm_clk_dev, (clock_control_subsys_t)0, &rate);
+	if (ret < 0) {
+		LOG_ERR("Failed to get camera external clock rate, error: (%d)", ret);
+		return ret;
+	}
+
+	LOG_INF("Camera external clock rate: (%u) Hz", rate);
+
+	return 0;
+}
+
+SYS_INIT(eth_pwm_clock_enable, POST_KERNEL, CONFIG_CLOCK_CONTROL_PWM_INIT_PRIORITY);
```

Comment:
have a common .c file with all of the code functions then use a board hook to run a function which runs them

Reply:
I removed the USB init file (which was a dirty solution anyway), will apply https://github.com/zephyrproject-rtos/zephyr/pull/85294 whrn it gets merged